mailpoet/lib/API/MP/v1/Subscribers.php

```diff
@@ -223,4 +186,92 @@ protected function _sendConfirmationEmail(SubscriberEntity $subscriberEntity) {
       );
     }
   }
+
+  /**
+   * @throws APIException
+   */
+  private function findSubscriber($subscriberIdOrEmail): SubscriberEntity {
+    if (empty($subscriberIdOrEmail)) {
+      throw new APIException(__('A subscriber is required.', 'mailpoet'), APIException::SUBSCRIBER_NOT_EXISTS);
+    }
+    // throw exception when subscriber does not exist
+    $subscriber = null;
+    if (is_int($subscriberIdOrEmail) || (string)(int)$subscriberIdOrEmail === $subscriberIdOrEmail) {
+      $subscriber = $this->subscribersRepository->findOneById($subscriberIdOrEmail);
+    } else if (strlen(trim($subscriberIdOrEmail)) > 0) {
+      $subscriber = $this->subscribersRepository->findOneBy(['email' => $subscriberIdOrEmail]);
+    }
+
+    if (!$subscriber) {
+      throw new APIException(__('This subscriber does not exist.', 'mailpoet'), APIException::SUBSCRIBER_NOT_EXISTS);
+    }
+
+    return $subscriber;
+  }
+
+  /**
+   * @return SegmentEntity[]
+   * @throws APIException
+   */
+  private function getAndValidateSegments(array $listIds, string $context = self::CONTEXT_SUBSCRIBE): array {
```

Comment:
Do you think it might be good to remove the default value for the `$context`? It looks like you're explicitly including the value everywhere this method is invoked, which I prefer to relying on default values. No big deal either way though.

Reply:
Good point. I'll remove it.